project_duplicated_task/models/project_task.py

```diff
@@ -0,0 +1,113 @@
+# Copyright 2021 Akretion (https://www.akretion.com).
+# @author Kévin ROche <kevin.roche@akretion.com>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class ProjectTask(models.Model):
+    _inherit = "project.task"
+
+    task_duplicated_open_ids = fields.Many2many(
+        comodel_name="project.task",
+        relation="duplicated_open_rel",
+        column1="duplicated_open_id",
+        column2="task_duplicated_id",
+        string="Duplicate Open Task",
+        store=True,
+        ondelete="restrict",
+    )
+
+    task_duplicated_closed_ids = fields.Many2many(
+        comodel_name="project.task",
+        relation="duplicated_closed_rel",
+        column1="duplicated_closed_id",
+        column2="task_duplicated_id",
+        string="Duplicate Closed Task",
+        readonly=True,
+        store=True,
+        ondelete="restrict",
+    )
+
+    task_duplicated_id = fields.Many2one(comodel_name="project.task", string="Task")
+
+    is_duplicate = fields.Boolean(default=False)
+    have_duplicate = fields.Boolean(default=False)
```

Comment:
```suggestion
    is_duplicate = fields.Boolean(default=False)
    have_duplicate = fields.Boolean(default=False)
```
should be computed field if really needed